### PR TITLE
Added Omen Spontaneous Combustion and light tube and mirror effects

### PR DESCRIPTION
**Pull request by carlarctg**
## About The Pull Request

Cursed crewmembers can randomly, extremely rarely, spontaneously combust for no reason.

Cursed crewmembers can get zapped by nearby light tubes.

Cursed crewmembers can freak out when passing by mirrors.

To make up for these, triggering a cursed effect is slightly less than half as likely now when walking around now.

## Why It's Good For The Game

Cursed is fun as hell, but after a certain point it gets kind of monotonous - it's airlocks, vending machines, and the rest is too rare to count. We need more ways to comically get hurt in the game.

You might dislike the 'reduced effects' bit but trust me it is incredibly frickin' common to have shit happen to you. Add to the occasional vending machine and airlock crushes the near-constant light tubes all over the station? Yeah, that needs a toning down else it will be just a tad too miserable to be funny. Also cause the poor janitor unneeded stress.

## Changelog

:cl:
add: Cursed crewmembers can randomly, extremely rarely, spontaneously combust for no reason.
add: Cursed crewmembers can get zapped by nearby light tubes.
add: Cursed crewmembers can freak out when passing by mirrors.
add: To make up for these, triggering a cursed effect is slightly less than half as likely now when walking around now.
/:cl:

### CODE REVIEW

File: code/datums/components/omen.dm [hunk 1]
```diff
@@ -65,7 +66,17 @@
 		return
 
 	var/mob/living/living_guy = our_guy
-	if(!prob(15 * luck_mod))
+
+	if((rand(10000) == 10000) && (living_guy.stat != DEAD)) // You hit the lottery! Kinda.
```
Comment (MrMelbert) [inline code/datums/components/omen.dm:70]:
```suggestion
	if(prob(0.001) && (living_guy.stat != DEAD)) // You hit the lottery! Kinda.
```

Comment (carlarctg) [reply code/datums/components/omen.dm]: Wasn't sure if prob handled decimals

File: code/datums/components/omen.dm [hunk 2]
```diff
@@ -65,7 +65,17 @@
 		return
 
 	var/mob/living/living_guy = our_guy
-	if(!prob(15 * luck_mod))
+
+	if(prob(0.001) && (living_guy.stat != DEAD)) // You hit the lottery! Kinda.
+		living_guy.visible_message(span_danger("[living_guy] suddenly bursts into flames!"), span_danger("You suddenly burst into flames!"))
+		INVOKE_ASYNC(living_guy, TYPE_PROC_REF(/mob, emote), "scream")
+		living_guy.adjust_fire_stacks(6)
```
Comment (Ghommie) [inline code/datums/components/omen.dm:72]:
```suggestion
		living_guy.adjust_fire_stacks(20)
```
It's the frikkin jackpot, come on. :joy:

File: code/datums/components/omen.dm [hunk 3]
```diff
@@ -104,6 +114,58 @@
...
+
+			living_guy.set_jitter_if_lower(25 SECONDS)
+			if(prob(7 * luck_mod))
+				to_chat(living_guy, span_warning("You are completely shocked by this turn of events!"))
+				var/mob/living/carbon/carbon_guy = living_guy
+				to_chat(living_guy, span_userdanger("You clutch at your heart!"))
+				if(istype(carbon_guy))
+					carbon_guy.set_heartattack(status = TRUE)
+
+			if(!permanent && !prob(66.6))
```
Comment (Time-Green) [inline code/datums/components/omen.dm:166]:
```suggestion
			if(!permanent && prob(33.3))
```

**Review by Time-Green — CHANGES_REQUESTED** [review]

File: code/datums/components/omen.dm [hunk 4]
```diff
@@ -104,6 +114,58 @@
...
+			evil_light.Beam(living_guy, icon_state = "lightning[rand(1,12)]", time = 0.5 SECONDS)
+			living_guy.electrocute_act(35 * (damage_mod / 2), evil_light, flags = SHOCK_NOGLOVES)
+			INVOKE_ASYNC(living_guy, TYPE_PROC_REF(/mob, emote), "scream")
+			if(!permanent && !prob(66.6))
+				qdel(src)
+
+		for(var/obj/structure/mirror/evil_mirror in the_turf)
+			to_chat(living_guy, span_warning("You pass by the mirror and glance at it..."))
+			if(evil_mirror.broken)
+				to_chat(living_guy, span_warning("You feel lucky, somehow."))
```
[the remark this answers is not in the document]
Comment (carlarctg) [reply code/datums/components/omen.dm:136]:
```suggestion
				to_chat(living_guy, span_notice("You feel lucky, somehow."))
```

**Pull request merged**